File: AirshipKit/AirshipKit/common/UAAppForegroundEvent.m
/* Copyright 2010-2019 Urban Airship and Contributors */

#import "UAAppForegroundEvent+Internal.h"
#import "UAEvent+Internal.h"

@implementation UAAppForegroundEvent

- (NSMutableDictionary *)gatherData:(UAUserData *)userData {
    NSMutableDictionary *data = [super gatherData:userData];
    [data removeObjectForKey:@"foreground"];
    return data;
}

- (NSString *)eventType {
    return @"app_foreground";
}

@end

File: Airship/AirshipCore/Source/UAAppForegroundEvent.swift
/* Copyright Airship and Contributors */

/**
 * @note For Interrnal use only :nodoc:
 */
@objc
public class UAAppForegroundEvent : UAAppInitEvent {
    public override func gatherData() -> [AnyHashable : Any] {
        var data = super.gatherData()
        data.removeValue(forKey: "foreground")
        return data
    }

    public override var eventType: String {
        get {
            return "app_foreground"
        }
    }
}
